applications/IgaApplication/custom_conditions/support_penalty_condition.cpp

```diff
@@ -0,0 +1,176 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ricky Aristio
+//                   Tobias Teschemacher
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_conditions/support_penalty_condition.h"
+
+namespace Kratos
+{
+    void SupportPenaltyCondition::CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag
+    )
+    {
+        KRATOS_TRY
+        const double penalty = GetProperties()[PENALTY_FACTOR];
+
+        const auto& r_geometry = GetGeometry();
+        const SizeType number_of_nodes = r_geometry.size();
+        const SizeType mat_size = r_geometry.WorkingSpaceDimension() * number_of_nodes;
+
+        // Integration
+        const GeometryType::IntegrationPointsArrayType& integration_points = r_geometry.IntegrationPoints();
+
+        // initial determinant of jacobian 
+        Vector determinat_jacobian_vector_initial(integration_points.size());
+        DeterminantOfJacobianInitial(r_geometry, determinat_jacobian_vector_initial);
+
+        for (IndexType point_number = 0; point_number < integration_points.size(); point_number++)

```

Comment:
i would use pre-increment instead of post-increment.

Reply:
👍